### DISCUSSION TITLE
Expand Archives with content more than 255 characters in Filename with a little Function

### DISCUSSION THREAD

**Discussion by StHoBT** [PSAppDeployToolkit/PSAppDeployToolkit#775] [Ideas]
Problem:
for packed archives with contained filenames longer than 255 characters the normal unpack routine aborts.
e.g.: Cygwin package

Solution:
To work around the problem I created a small function that unpacks the data correctly using 7zip.

I can provide this solution with pleasure.

Here is the function description:
Function Expand-ArchiveCustom{
<#
.SYNOPSIS
	Expand a Archives like 7z, XZ, BZIP2, GZIP, TAR, ZIP, WIM R, ARJ, CAB, CHM, CPIO, CramFS, DMG, FAT, HFS, IHEX, ISO, LZH, LZMA, MBR, MSI, NSIS, NTFS, RAR, RPM, SquashFS, UDF, UEFI, VHD, WIM, XAR and Z to a Destinationfolder
.DESCRIPTION
	Expand a Archives like 7z, XZ, BZIP2, GZIP, TAR, ZIP, WIM R, ARJ, CAB, CHM, CPIO, CramFS, DMG, FAT, HFS, IHEX, ISO, LZH, LZMA, MBR, MSI, NSIS, NTFS, RAR, RPM, SquashFS, UDF, UEFI, VHD, WIM, XAR and Z to a Destinationfolder
.SYNTAX
    Expand-ArchiveCustom [-archiveFilePath] <String> [-destinationPath] <String>] [[-password] <String>] [[-Overwrite [boolean]] [[-IgnoreExitCodes <String>]] [[-PassThru [boolean]] [[-Debuglevel] [int]] 
.PARAMETER archiveFilePath
	the String of the Archive-File-Path that is to expand. -archiveFilePath "C:\temp\TestSHO.zip"
.PARAMETER destinationPath
	the String of the destinationPath. -destinationPath "C:\temp"
.PARAMETER Overwrite
	Specifies that all existing Files on disk will be overwrite
.PARAMETER IgnoreExitCodes
	List the exit codes to ignore.
.PARAMETER PassThru
	Returns ExitCode, STDOut, and STDErr output from the process.
.PARAMETER Debuglevel
	if the debuglevel is higher then 0 debuginformations will be written in the 'Bertrandt Custom Log'. The Parameter is optional. default=0
	0=no debug , 1=only ReturnValue (no ReturnValue is present) , 2=full debug with most values
.EXAMPLE
	$ReturnCode = Expand-ArchiveCustom "C:\temp\TestSHO.zip" "C:\temp" -Debuglevel 2
.NOTES
	RetunCode-Definition:
	11 -> The archive was not found.
	12 -> Destinationfolder or a valid parentfolder not found.
	13 -> Not enough free Discspace on destinationPath.
	14 -> Error expanding Archive.
	
	7Zip-Returncodes using Passthru
	0 -> No error 
	1 -> Warning (Non fatal error(s)). For example, one or more files were locked by some other application, so they were not compressed. 
	2 -> Fatal error 
	7 -> Command line error 
	8 -> Not enough memory for operation 
	255 -> User stopped the process 

	Attention: You must use the Full ´Path-String. Asterisk in Path are not allowed
	Author of this feature: Steffen Hoffmann
#>

**Comment by cyborgrd** [comment]
Please use "Add Code" after copy /paste and marking, otherwise your scripts are broken.

**Comment by mjr4077au** [comment]
Anything depending on 3rd party binaries regrettably won't be included.

**Comment by DanGough** [comment]
The solution here should really be to enable long file names on the clients:

https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later
